File: php/src/Queue/Queue.php
<?php

namespace Claudiosilvajunior\Php\Queue;

class Queue
{
    private array $queue = [];
    private int $lengthQueue = 0;

    public function push(mixed $element): void
    {
        $this->queue[] = $element;
        $this->lengthQueue++;
    }

    public function empty(): bool
    {
        return $this->lengthQueue === 0;
    }

    public function pop(): void
    {
        if ($this->empty()) {
            return;
        }

        array_shift($this->queue);
        $this->lengthQueue--;
    }

    public function length(): int
    {
        return $this->lengthQueue;
    }

    public function front(): mixed
    {
        if ($this->empty()) {
            return null;
        }

        return $this->queue[0];
    }
}

File: java/src/main/java/org/example/queue/Queue.java
package org.example.queue;

import java.util.ArrayList;
import java.util.List;

public class Queue {
    private final List<String> queue;
    private int length = 0;

    public Queue() {
        queue = new ArrayList<>();
    }

    public Boolean empty() {
        return queue.isEmpty();
    }

    public int length() {
        return length;
    }

    public void push(String element) {
        queue.add(element);
        length++;
    }

    public void pop() {
        if (empty()) {
            return;
        }

        queue.remove(0);
        length--;
    }

    public String front() {
        return queue.get(0);
    }
}
